src/vue/modules/loyalty-points-statistics/components/charts/pie-chart.vue

```diff
@@ -0,0 +1,234 @@
+
+      this.svg.select('.pie-chart__slices')
+        .datum(this.data)
+        .selectAll('path')
+        .data(pie)
+        .enter().append('path')
+        .attr('fill', item => item.data.color)
+        .attr('d', this.innerArc)
+    },
+
+    renderLabel (pie) {
+      this.svg.append('g').attr('class', 'pie-chart__label')
+
+      this.svg.select('.pie-chart__label')
+        .datum(this.data)
+        .selectAll('text')
+        .data(pie)
+        .enter()
+        .append('text')
+        .attr('dy', '.35em')
+        .html(item => `${item.data.label}: <tspan>${item.data.value}</tspan>`)
+        .attr('transform', item => {
+          const labelPosition = SCALE_COEFF * this.radius
+          let pos = this.outerArc.centroid(item)
+
+          pos[0] = labelPosition * this.getMidAngleDirectionSign(item)
+          return `translate(${pos})`
+        })
+        .style('text-anchor', item => {
+          return this.getMidAngleDirectionSign(item) > 0 ? 'start' : 'end'
+        })
+    },
+
+    renderLines (pie) {
+      this.svg.append('g').attr('class', 'pie-chart__lines')
+
+      this.svg.select('.pie-chart__lines')
+        .datum(this.data)
+        .selectAll('polyline')
+        .data(pie)
+        .enter()
+        .append('polyline')
+        .attr('points', item => {
+          const linePosition = (SCALE_COEFF - 0.05) * this.radius
+          let pos = this.outerArc.centroid(item)
+
+          pos[0] = linePosition * this.getMidAngleDirectionSign(item)
+
+          return [
+            this.innerArc.centroid(item),
+            this.outerArc.centroid(item),
+            pos,
+          ]
+        })
+    },
+
+    renderToolTip (selection) {
+      const tipMarginTop = 10
+      const tipFontSize = '2.4rem'
+      const tipOpacity = 0.35
+
+      selection.on('mouseenter', item => {
+        this.svg.append('text')
+          .attr('class', 'pie-chart__tool-circle')
+          .attr('dy', tipMarginTop)
+          .html(this.percentFormat(item.data.value / this.totalValue))
+          .style('font-size', tipFontSize)
+          .style('text-anchor', 'middle')
+
+        this.svg.append('circle')
+          .attr('class', 'pie-chart__tool-circle')
+          .attr('r', this.radius * this.innerRadius * SCALE_COEFF)
+          .style('fill', item.data.color)
+          .style('fill-opacity', tipOpacity)
+      })
+
+      selection.on('mouseout', () => {
+        d3.selectAll('.pie-chart__tool-circle').remove()
+      })
+    },
+
+    getMidAngleDirectionSign (item) {
+      const midAngle = item.startAngle + (item.endAngle - item.startAngle) / 2
+      return midAngle < Math.PI ? 1 : -1
+    },
+  },
+}
+</script>
+
+<style lang="scss">
+@import "~@scss/variables";
+
+.pie-chart {
+  svg {
+    filter: drop-shadow(0rem 0.3rem 0.3rem $col-pie-chart-shadow);
+  }
+
+  &__lines polyline {
+    opacity: 0.3;
+    stroke-width: 0.2rem;
+    stroke: $col-pie-chart-stroke;
+    fill: none;
+  }
+
+  &__label {
```

Comment:
Please, don't use this (&) with classes. This is a good feature, but it makes a search by class name a bit hard. In this case, you cannot search for your `pie-chart__lines` class by the application.

Reply:
Also, the label's styles look like copy-pasted from other resource and does not fit into our design

Reply:
Updated